interfaces/cython/cantera/composite.py

```diff
@@ -516,32 +516,38 @@ def __init__(self, phase, shape=(0,), states=None, extra=None):
         reserved = self.__dir__()
 
         self._extra = {}
-        if isinstance(extra, dict):
-            for name, v in extra.items():
+
+        if isinstance(extra, str):
+            extra = [extra]
+
+        if isinstance(extra, (list, tuple, np.ndarray)) and all(isinstance(name, str) for name in extra):
+            #if extra is a list and all elements in extra are of type str
+            for name in extra:
                 if name in reserved:
                     raise ValueError(
                         "Unable to create extra column '{}': name is already "
                         "used by SolutionArray objects.".format(name))
-                if not np.shape(v):
-                    self._extra[name] = [v]*self._shape[0]
-                elif len(v) == self._shape[0]:
-                    self._extra[name] = list(v)
-                else:
-                    raise ValueError("Unable to map extra SolutionArray"
-                                     "input for named {!r}".format(name))
+                self._extra[name] = np.empty(self._shape)
 
-        elif extra and self._shape == (0,):
-            for name in extra:
+        elif isinstance(extra, dict):
+            for name, v in extra.items():
                 if name in reserved:
                     raise ValueError(
                         "Unable to create extra column '{}': name is already "
                         "used by SolutionArray objects.".format(name))
-                self._extra[name] = []
+                if not np.shape(v):
+                    self._extra[name] = np.array([v]*self._shape)
+                elif len(v) == self._shape or np.array(v).shape == self._shape:
+                    self._extra[name] = np.array(list(v))
+                else:
+                    raise ValueError("Unable to map extra SolutionArray"
+                                     "input for named {!r}".format(name))
 
         elif extra:
             raise ValueError("Initial values for extra properties must be"
-                " supplied in a dict if the SolutionArray is not initially"
-                " empty")
+                " supplied in a dict or can be initialised with the property string"
```

Comment:
I'd like to improve this error message. What about:

> Extra properties can be created by passing a sequence of names for the properties, if the SolutionArray is not initially empty. If you want to supply initial values for the properties, use a dictionary whose keys are the names of the properties and values are the initial values.

Does that make sense and still capture what we're going for here?

Reply:
this is certainly better but does `sequence` imply an iterable of strings? I don't know about this

Reply:
A sequence implies any iterable, at least to me, but maybe `iterable` would be a better word there instead of `sequence`.